spec-tests/src/test/java/org/candlepin/spec/content/ContentResourceSpecTest.java

```diff
@@ -0,0 +1,286 @@
+        String firstContentName = jsonBody.get("products").get(0).get("content")
+            .get(0).get("name").asText();
+        if (firstContentName.equals(content1.getName())) {
+            contentOutput1 = jsonBody.get("products").get(0).get("content").get(0);
+            contentOutput3 = jsonBody.get("products").get(0).get("content").get(1);
+        }
+        else {
+            contentOutput1 = jsonBody.get("products").get(0).get("content").get(1);
+            contentOutput3 = jsonBody.get("products").get(0).get("content").get(0);
+        }
+
+        assertEquals(content1.getType(), contentOutput1.get("type").asText());
+        assertEquals(content1.getName(), contentOutput1.get("name").asText());
+        assertEquals(content1.getLabel(), contentOutput1.get("label").asText());
+        assertEquals(content1.getVendor(), contentOutput1.get("vendor").asText());
+        assertEquals(content1.getContentUrl(), contentOutput1.get("path").asText());
+        assertEquals(content1.getArches(), contentOutput1.get("arches").get(0).asText());
+
+        assertEquals(content3.getType(), contentOutput3.get("type").asText());
+        assertEquals(content3.getName(), contentOutput3.get("name").asText());
+        assertEquals(content3.getLabel(), contentOutput3.get("label").asText());
+        assertEquals(content3.getVendor(), contentOutput3.get("vendor").asText());
+        assertEquals(content3.getContentUrl(), contentOutput3.get("path").asText());
+        assertEquals(0, contentOutput3.get("arches").size());
+    }
+
+    @Test
+    @DisplayName("should filter content with mismatched architecture from the product")
+    public void filterContentWithMismatchedArchitectureFromProduct() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        // Even though this product has no arches specified, and should normally not be filtered out,
+        // the product it belongs to has an architecture that mismatches with the system's,
+        // so we do expect it to get filtered out.
+        ContentDTO content1 = createContent(owner.getKey(), StringUtil.random("cname1"),
+            StringUtil.random("test-content1"), StringUtil.random("clabel1"), StringUtil.random("ctype1"),
+            StringUtil.random("cvendor1"), "/this/is/the/path", null);
+        AttributeDTO attribute1 = new AttributeDTO();
+        attribute1.setName("arch");
+        attribute1.setValue("x86_64");
+        ProductDTO product1 = createProduct(StringUtil.random("product"), owner.getKey(), null, attribute1);
+        product1 = ownerProductApi.addContent(owner.getKey(), product1.getId(), content1.getId(), true);
+
+        // This content has no arches specified, but the product it belongs to has an arch that
+        // matches with that of the system's, so we do NOT expect it to get filtered out.
+        ContentDTO content2 = createContent(owner.getKey(), StringUtil.random("cname2"),
```

Comment:
Could you add this to Contents.random()? It would be great having a factory method for randomized contents.

Reply:
Added Contents class and added random method for generating random contents